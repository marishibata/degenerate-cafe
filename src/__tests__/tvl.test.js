import React from 'react';
import { render } from '@testing-library/react';
import TVL from '../Components/tvl';

jest.mock('axios', () => {
  return {
    default: () => {
      return new Promise((resolve, reject) => {
        resolve({
          data:  [
            {
              id: "1",
              name: "Uniswap",
              tvl: 100000,
              change_1d: -100
            },
          ]
        })
      })
    }
  }
});


describe('TVL section', () => {
  it('Renders consistently', async () => {
    const { asFragment } = render(<TVL/>);
    expect(asFragment()).toMatchSnapshot();
  })


  it('Successfully renders the TVL component', async () => {
    const { findByText } = render(<TVL/>);
    const result = await findByText('DeFi Protocol Total Value Locked');
    expect(result).toBeTruthy();
  })

  it('Successfully fetches TVL data', async () => {
    const { findByText } = render(<TVL/>);
    const result = await findByText('Uniswap');
    expect(result).toBeTruthy();
  })
})