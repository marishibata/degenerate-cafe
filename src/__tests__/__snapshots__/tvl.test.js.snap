// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TVL section Renders consistently 1`] = `
<DocumentFragment>
  <div
    class="tvl-parent"
  >
    <div
      class="title"
    >
      <span>
        DeFi Protocol Total Value Locked
      </span>
    </div>
    <div
      class="tvl-item-header"
    >
      <div
        class="tvl1"
      >
        Procotol
      </div>
      <div
        class="tvl2"
      />
      <div
        class="tvl3"
      >
        TVL
      </div>
      <div
        class="tvl4"
      >
        Daily % Cx
      </div>
      <div
        class="tvl5"
      />
      <div
        class="tvl6"
      >
        Escan
      </div>
    </div>
    <div
      class="tvl-data"
    />
  </div>
</DocumentFragment>
`;
